src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,26 @@
 import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from '../../app/store';
+import { setQuery, setStatus } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const { status, query } = useSelector((state: RootState) => state.filter);
+  const dispatch: AppDispatch = useDispatch();
+
+  const handleStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setStatus(event.target.value as Status));
+  };
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(setQuery(event.target.value));
+  };
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select data-cy="statusSelect" value={status} onChange={handleStatus}>
```

Comment:
Ensure that the `value` attribute of the `<select>` element corresponds to a valid `Status` type. If `status` is not a valid `Status`, it may cause issues.